services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -6526,6 +6524,11 @@ public void killOfflinePush(String clusterName, String kafkaTopic, boolean isFor
       LOGGER.info("Resource: {} doesn't exist, kill job will be skipped", kafkaTopic);
       return;
     }
+
+    deleteHelixResource(clusterName, kafkaTopic);
```

Comment:
This seems like a harmful change because it deletes helix resources without validating if it's not the current version, etc. Can you share some more details about the issue that was noticed?

Reply:
Everytime a VPJ runs the `killOfflinePush` job, it does not remove the helix resources. As part of the job, it calls the killOfflinePush but does not call the method to remove the helix resource. 

Example of it doing the killOfflinePush & removing helix resource correctly : https://github.com/linkedin/venice/blob/main/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java#L3276

https://github.com/linkedin/venice/blob/main/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java#L3274

I see that we do have version validates & i will add this to the deleteHelixResource()

Reply:
You need to query the child controllers to validate none of them are using the killjob topic as the current version.
Also I am thinking what is the harm in keeping the helix resources intact till the time when the store versions are deleted (which happens later in some background thread).

Reply:
@majisourav99  made the changes requested. Pl lmk if this looks ok. 

also, in another [location](https://github.com/linkedin/venice/pull/1069/files#diff-36114654b45af0adf03521759f0301f267c4b453321c47a4d1c1c4726abddca2L3273), it doesn't keep helix resource intact until store versions are deleted. Is it ok to follow the same here?